test/HTS.t.sol

```diff
@@ -141,4 +143,130 @@ contract HTSTest is Test, TestSetup {
         HtsSystemContract(token).getTokenInfo(token);
     }
+
+    function test_mintToken_should_succeed_with_valid_input() external {
+        if (TestMode.JSON_RPC == testMode) {
+            // TODO: skip this test until the hardhat solution for getTokenInfo is implemented
+            return;
+        }
+
+        address token = USDC;
+        address treasury = USDC_TREASURY;
+        int64 amount = 1000;
+        int64 initialTotalSupply = 10000000005000000;
+        uint256 initialTreasuryBalance = IERC20(token).balanceOf(treasury);
+        bytes[] memory metadata = new bytes[](0);
+
+        (int64 responseCode, int64 newTotalSupply, int64[] memory serialNumbers) = HtsSystemContract(HTS_ADDRESS).mintToken(token, amount, metadata);
+        assertEq(responseCode, 22);
+        assertEq(serialNumbers.length, 0);
+        assertEq(newTotalSupply, initialTotalSupply + amount);
+        assertEq(IERC20(token).balanceOf(treasury), uint64(initialTreasuryBalance) + uint64(amount));
+    }
+
+    function test_mintToken_should_revert_with_invalid_treasureAccount() external {
+        address token = USDC;
+        int64 amount = 1000;
+        bytes[] memory metadata = new bytes[](0);
+        int64 initialTotalSupply = 10000000005000000;
+        IHederaTokenService.TokenInfo memory tokenInfo;
+        tokenInfo.token = IHederaTokenService.HederaToken(
+            "USD Coin",
+            "USDC",
+            address(0),
+            "USDC HBAR",
+            false,
+            initialTotalSupply + amount,
+            false,
+            new IHederaTokenService.TokenKey[](0),
+            IHederaTokenService.Expiry(0, address(0), 0)
+        );
+
+        vm.mockCall(token, abi.encode(HtsSystemContract.getTokenInfo.selector), abi.encode(22, tokenInfo));
+        vm.expectRevert(bytes("mintToken: invalid account"));
+        HtsSystemContract(HTS_ADDRESS).mintToken(token, amount, metadata);
+    }
+
+    function test_mintToken_should_revert_with_invalid_token() external {
+        address token = address(0);
+        int64 amount = 1000;
+        bytes[] memory metadata = new bytes[](0);
+
+        vm.expectRevert(bytes("mintToken: invalid token"));
+        HtsSystemContract(HTS_ADDRESS).mintToken(token, amount, metadata);
+    }
+
+    function test_mintToken_should_revert_with_invalid_amount() external {
+        address token = address(0x123);
+        int64 amount = 0;
+        bytes[] memory metadata = new bytes[](0);
+
+        vm.expectRevert(bytes("mintToken: invalid amount"));
+        HtsSystemContract(HTS_ADDRESS).mintToken(token, amount, metadata);
+    }
+
+    function test_burnToken_should_succeed_with_valid_input() external {
+        if (TestMode.JSON_RPC == testMode) {
+            // TODO: skip this test until the hardhat solution for getTokenInfo is implemented
+            return;
+        }
+
+        address token = MFCT;
+        address treasury = MFCT_TREASURY;
```

Comment:
I'm not very keen on this as we're relying on `getTokenInfo` to work properly and we could potentially hide a bug instead of verifying it through the test